newIDE/app/src/PropertiesEditor/PropertiesMapToSchema.js

```diff
@@ -114,6 +115,37 @@ export default (
         getLabel,
         getDescription,
       };
+    } else if (valueType === 'behavior') {
+      const behaviorType =
+        property.getExtraInfo().size() > 0 ? property.getExtraInfo().at(0) : '';
+      return {
+        name,
+        valueType: 'string',
+        getChoices: () => {
+          return !object || behaviorType === ''
+            ? []
+            : object
+                .getAllBehaviorNames()
+                .toJSArray()
+                .map(name =>
+                  object.getBehavior(name).getTypeName() === behaviorType
+                    ? name
+                    : null
+                )
+                .filter(Boolean)
+                .map(value => ({ value, label: value }));
+        },
```

Comment:
I still had to get the extraInfo outside because they are missing when it's refreshing. I wonder if there is an issue in the PropertiesEditor. I will investigate.

Reply:
Right this might be because we're dealing with libGD.js C++ objects here, i.e: objects living in the WebAssembly memory. While `object` will remain valid for the whole life of the component using the properties (because if the object is deleted, it's the IDE responsibility to close anything related to the object), it's not necessarily the case for some objects returned by some methods (those marked with `[Value]` in `Bindings.idl` will get replaced at the next time the function is called).

This being said, here I would have assumed `property` would remain valid... Anyway, it's better now! And safer to save the behaviorType in a traditional JS string :)